tests/test_db.py

```diff
@@ -110,3 +114,47 @@ def test_run_mysql_executemany_query_execute_error(config):
         # check connection is closed
         assert cursor.close.called == True
         assert conn.close.called == True
+
+def test_create_dart_sql_server_conn_readonly(config):
+    with patch('pyodbc.connect') as mock_connect:
+        conn_string = f'DRIVER={config.DART_DB_DRIVER};SERVER={config.DART_DB_HOST};PORT={config.DART_DB_PORT};DATABASE={config.DART_DB_DBNAME};UID={config.DART_DB_RO_USER};PWD={config.DART_DB_RO_PASSWORD}'
+        create_dart_sql_server_conn(config)
+        mock_connect.assert_called_with(conn_string)
+
+def test_create_dart_sql_server_conn_readwrite(config):
+    with patch('pyodbc.connect') as mock_connect:
+        conn_string = f'DRIVER={config.DART_DB_DRIVER};SERVER={config.DART_DB_HOST};PORT={config.DART_DB_PORT};DATABASE={config.DART_DB_DBNAME};UID={config.DART_DB_RW_USER};PWD={config.DART_DB_RW_PASSWORD}'
+        create_dart_sql_server_conn(config, readonly=False)
+        mock_connect.assert_called_with(conn_string)
+
+def test_create_dart_sql_server_conn_none(config):
+    with patch('pyodbc.connect', return_value = None):
+        assert create_dart_sql_server_conn(config) == None
+
+def test_create_dart_sql_server_conn_expection(config):
+    with patch('pyodbc.connect', side_effect = Exception('Boom!')):
+        with pytest.raises(Exception):
+            create_dart_sql_server_conn(config)
+
```

Comment:
Huh... okay, looks like I might have misunderstood python exception handling if this test is passing.

Reply:
Yeah this looks like it should not raise an exception but should instead return `None`. Will investigate

Reply:
The try/except was catching an `Error` where I think it should have been catching an `Exception`. Will check which was desired and change as appropriate

Reply:
Updated to explicitly catch mysql.Errors and pyodbc.Errors where appropriate